x-pack/test/detection_engine_api_integration/utils/preview_rule_with_exception_entries.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ToolingLog } from '@kbn/tooling-log';
+import type SuperTest from 'supertest';
+import type { NonEmptyEntriesArray, OsTypeArray } from '@kbn/securitysolution-io-ts-list-types';
+import type {
+  CreateRulesSchema,
+  RulePreviewLogs,
+} from '@kbn/security-solution-plugin/common/detection_engine/schemas/request';
+
+import { createContainerWithEntries } from './create_container_with_entries';
+import { createContainerWithEndpointEntries } from './create_container_with_endpoint_entries';
+import { previewRule } from './preview_rule';
+
+/**
+ * Convenience testing function where you can pass in just the entries and you will
+ * get a rule created with the entries added to an exception list and exception list item
+ * all auto-created at once.
+ * @param supertest super test agent
+ * @param rule The rule to create and attach an exception list to
+ * @param entries The entries to create the rule and exception list from
+ * @param endpointEntries The endpoint entries to create the rule and exception list from
+ * @param osTypes The os types to optionally add or not to add to the container
+ */
+export const previewRuleWithExceptionEntries = async ({
+  supertest,
+  log,
+  rule,
+  entries,
+  endpointEntries,
+  invocationCount,
+  timeframeEnd,
+}: {
+  supertest: SuperTest.SuperTest<SuperTest.Test>;
+  log: ToolingLog;
+  rule: CreateRulesSchema;
+  entries: NonEmptyEntriesArray[];
+  endpointEntries?: Array<{
+    entries: NonEmptyEntriesArray;
+    osTypes: OsTypeArray | undefined;
+  }>;
+  invocationCount?: number;
+  timeframeEnd?: Date;
+}): Promise<{
```

Comment:
nit: return type here not needed as it will be the same as return type result of called `previewRule`